src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/ActiveDebugFrameworkServices.cs

```diff
@@ -0,0 +1,105 @@
+
+            if(!string.IsNullOrWhiteSpace(targetFrameworks))
+            {
+                return TargetFrameworkProjectConfigurationDimensionProvider.ParseTargetFrameworks(targetFrameworks).ToList();
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// <see cref="IActiveDebugFrameworkServices.SetActiveDebuggingFrameworkPropertyAsync"/>
+        /// </summary>
+        public async Task SetActiveDebuggingFrameworkPropertyAsync(string activeFramework)
+        {
+            var props = await CommonProjectServices.ActiveConfiguredProjectProperties.GetProjectDebuggerPropertiesAsync().ConfigureAwait(false);
+            await props.ActiveDebugFramework.SetValueAsync(activeFramework).ConfigureAwait(true);
+        }
+
+        /// <summary>
+        /// <see cref="IActiveDebugFrameworkServices.GetActiveDebuggingFrameworkPropertyAsync"/>
+        /// </summary>
+        public async Task<string> GetActiveDebuggingFrameworkPropertyAsync()
+        {
+            var props = await CommonProjectServices.ActiveConfiguredProjectProperties.GetProjectDebuggerPropertiesAsync().ConfigureAwait(false);
+            var activeValue = await props.ActiveDebugFramework.GetValueAsync().ConfigureAwait(true) as string;
+            return activeValue;
+        }
+
+        /// <summary>
+        /// <see cref="IActiveDebugFrameworkServices.GetConfiguredProjectForActiveFrameworkAsync"/>
+        /// </summary>
+        public async Task<ConfiguredProject> GetConfiguredProjectForActiveFrameworkAsync()
+        {
+            var configProjects = await ActiveConfiguredProjectsProvider.GetActiveConfiguredProjectsMapAsync().ConfigureAwait(false);
+
+            // If there is only one we are done
+            if(configProjects.Count == 1)
+            {
+                return configProjects.First().Value;
+            }
+            
+            var activeFramework = await GetActiveDebuggingFrameworkPropertyAsync().ConfigureAwait(false);
+            if(!string.IsNullOrWhiteSpace(activeFramework))
+            {
+                if(configProjects.TryGetValue(activeFramework, out ConfiguredProject configuredProject))
+                {
+                    return configuredProject;
+                }
+            }
+
+            // We can't just select the first one. If activeFramework is not set we must pick the first one as defined by the 
+            // targetFrameworks property. So we need the order as returned by GetProjectFrameworks()
+            var frameworks = await GetProjectFrameworksAsync().ConfigureAwait(false);
+            if(frameworks != null &&  frameworks.Count > 0)
+            {
+                if(configProjects.TryGetValue(frameworks[0], out ConfiguredProject configuredProject))
+                {
+                    return configuredProject;
+                }
+
+            }
+
+            // All that is left is to return the first one.
```

Comment:
When do we get to this state? It seems like we would get here if TargetFramework or TargetFrameworks is not mentioned. Which would mean configProjects will be empty we will throw an exception at `configProjects.First()`